DepositRegistrar.sol

```diff
@@ -0,0 +1,306 @@
+// Name registrar using deposits
+
+contract Deed {
+    // The Deed is a contract intended simply to hold ether
+    // It can be controlled only by the registrar and can only send ether back to the owner
+    address public owner;
+    address public bidder;
+    Registrar public registrar;
+    bytes32 public sealedBid;
+    bytes32 public hashedName;
```

Comment:
If you just have users submit `sha3('foo')` instead of `sha3('foo.eth')`, then you can pass this hashed token directly to the ENS registrar; no need to deal with unhashed names at all.

In DNS terminology, `foo.bar` is a domain, while `foo` and `bar` are labels; there's no convention for `H('foo')`, but I've just been calling them labels, since we don't use the unhashed ones anywhere.